doc/design/diag_attribute_error.md

```diff
@@ -0,0 +1,107 @@
+# Diagnostic Attribute Error
+
+## Motivation
+
+SQLFlow extended SQL syntax to describe an end-to-end machine learning job, for a typical SQL program:
+
+``` python
+SELECT * FROM iris.train
+TO TRAIN DNNClassifier
+WITH
+  model.n_classes = 2,
+  model.hidden_units = [128, 64]
+LABEL class
+INTO my_model;
+
+SELECT * FROM iris.test
+TO PREDICT iris.pred.class
+USING my_model;
+```
+
+SQLFlow compiles each statement in the program into an execution plan and executes them.
+As the TRAIN statement above, SQLFlow uses `TO TRAIN` clause to train a specific model called `DNNClassifier`,
+using `WITH` clause to configure the training arguments.
+
+Sometimes users may make some configuration mistake on `WITH` clause,
+then the job would fail during execution and return some uncertain error message.
+
+The model parameter documentation describes parameters and acceptable values of human reading.
+We want to enhance it for the reading by the SQLFlow compiler so to warn about wrongly set parameters, and
+this can active three advantages at least:
+
+1. Early testing, we can do early testing before running the job; users can wait less time and cluster save resources.
```

Comment:
I am afraid that for security reasons, we cannot early check the values because we can only run user code (in either docstring or in Python source files) in the step containers, but not the SQLFlow server container?

Reply:
Yes, but for the built-in models(Tensorflow Estimator and XGBoost ), I think we can still check them on the SQLFlow server because we can review the code.